Marlin/Marlin_main.cpp

```diff
@@ -9993,13 +9993,13 @@ void setup() {
   SERIAL_ECHO_START;
 
   // Check startup - does nothing if bootloader sets MCUSR to 0
-  byte mcu = MCUSR;
+  byte mcu = HAL_get_reset_source();
   if (mcu & 1) SERIAL_ECHOLNPGM(MSG_POWERUP);
   if (mcu & 2) SERIAL_ECHOLNPGM(MSG_EXTERNAL_RESET);
   if (mcu & 4) SERIAL_ECHOLNPGM(MSG_BROWNOUT_RESET);
   if (mcu & 8) SERIAL_ECHOLNPGM(MSG_WATCHDOG_RESET);
   if (mcu & 32) SERIAL_ECHOLNPGM(MSG_SOFTWARE_RESET);
-  MCUSR = 0;
+  HAL_clear_reset_source();
```

Comment:
Other CPUs might not have the reset reasons in these bit patterns. A HAL_report_reset_reason() is probably the better solution for this. AVR would then parse the MCUSR Bytes. And others can do different things,..